src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/RGGameAction.cs

```diff
@@ -27,15 +21,24 @@ public abstract class RGGameAction
          /// The range of parameter values accepted by this action's
          /// Perform() method.
         public abstract IRGValueRange ParameterRange { get; }
-        
-        /// User-specified data for this action.
-        public Dictionary<string, object> UserData { get; }
-        
-        public RGGameAction(string[] path, Type objectType, int actionGroup)
+
+         /// The name of this action as it should be displayed when presented to the user.
+        public abstract string DisplayName { get; }
+
+        public RGGameAction(string[] path, Type objectType)
         {
             Paths = new List<string[]> { path };
             ObjectType = objectType;
-            ActionGroup = actionGroup;
+        }
+
+        public RGGameAction(RGSerializedAction serializedAction)
+        {
+            Paths = new List<string[]>(serializedAction.paths.Select(path => path.Split("/")));
+            ObjectType = Type.GetType(serializedAction.objectTypeName);
```

Comment:
This needs the second boolean argument to avoid throwing an exception on not found.

```suggestion
            ObjectType = Type.GetType(serializedAction.objectTypeName, false);
```

Reply:
My thought on this is that I would actually prefer it to throw an exception, since an invalid type name signifies that the analysis results are no longer applicable to the current code base and the deserialization process should be aborted.

Here I have handling for any exceptions that come out of the deserialization, notifying that the analysis needs to be re-run if an exception is thrown.
https://github.com/Regression-Games/RGUnityBots/blob/dc42e2d139b369b4fffc86f50ff2875c00278aca/src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/RGActionProvider.cs#L48-L52

Reply:
if we're expecting it to throw the exception, then we can remove the check for null on the line below

Reply:
I've removed the null check and this will just throw an exception if the type isn't found now.